generators/generator-base-blueprint.js

```diff
@@ -84,51 +90,187 @@ module.exports = class extends BaseGenerator {
         }
         let useBlueprints = false;
 
-        const blueprints = jhipsterUtils.parseBluePrints(
-            this.options.blueprints ||
-                this.configOptions.blueprints ||
-                this.config.get('blueprints') ||
-                this.options.blueprint ||
-                this.configOptions.blueprint ||
-                this.config.get('blueprint')
-        );
-
-        if (blueprints && blueprints.length > 0) {
-            // Verify blueprints, should be executed only once
-            if (!this.configOptions.blueprintsVerified) {
-                // Run a lookup to find blueprints.
-                const packagePatterns = blueprints
-                    .filter(bp => !this.env.isPackageRegistered(jhipsterUtils.packageNameToNamespace(bp.name)))
-                    .map(bp => bp.name);
-                this.env.lookup({ filterPaths: true, packagePatterns });
-
-                if (!this.options.skipChecks) {
-                    const namespaces = blueprints.map(bp => jhipsterUtils.packageNameToNamespace(bp.name));
-                    // Verify if the blueprints has been registered.
-                    const missing = namespaces.filter(namespace => !this.env.isPackageRegistered(namespace));
-                    if (missing && missing.length > 0) {
-                        this.error(`Some blueprints were not found ${missing}, you should install them manually`);
-                    }
+        // Load and verify blueprints, should be executed only once
+        if (!this.configOptions.blueprintConfigured) {
+            this.configOptions.blueprintConfigured = true;
+            let argvBlueprints = this.options.blueprints || '';
+            // check for old single blueprint declaration
+            const blueprint = this.options.blueprint;
+            if (blueprint) {
+                this.warning('--blueprint option is deprecated. Please use --blueprints instead');
+                if (!argvBlueprints.split(',').includes(blueprint)) {
+                    argvBlueprints = `${blueprint},${argvBlueprints}`;
                 }
+            }
+            const blueprints = jhipsterUtils.mergeBlueprints(
+                jhipsterUtils.parseBluePrints(argvBlueprints),
+                jhipsterUtils.loadBlueprintsFromConfiguration(this.config)
+            );
+            // Run a lookup to find blueprints.
```

Comment:
if you need write comments, then it should be moved to another function with an explicit name :)

Reply:
Same as above.